upgrade/upgrade-scripts/upgrade-container

```diff
@@ -471,7 +471,7 @@ function migrate_file() {
 		parentdir="$(dirname "$path")"
 		mkdir -p "/var/lib/machines/${CONTAINER}${parentdir}" ||
 			die "'mkdir -p $parentdir' failed for '$CONTAINER'"
-		cp -p "$path" "/var/lib/machines/${CONTAINER}${path}" ||
+		cp -Pp "$path" "/var/lib/machines/${CONTAINER}${path}" ||
```

Comment:
We might want to also update the logic on line 572 for `resolve.conf` to check that it is not a symlink before calling `migrate_file` on it.

Reply:
I think the comment had good intentions, but the code was previously wrong. It looks like migration was always copying `/etc/resolv.conf` regardless of whether or not it was a symlink.
```
root@ip-10-110-235-164:~# ls -l /etc/default/nfs-kernel-server 
lrwxrwxrwx 1 root root 44 Jul 29 15:20 /etc/default/nfs-kernel-server -> /opt/delphix/server/etc/nfs-kernel-server-v4
root@ip-10-110-235-164:~# if [[ -f /etc/default/nfs-kernel-server ]]; then
> echo "yep, a regular file"
> fi
yep, a regular file
```